internal/executor/executor.go

```diff
@@ -190,28 +192,60 @@ func getExpandedScriptEnvironment(executor *Executor, responseEnvironment map[st
 	}
 	responseEnvironment["CIRRUS_OS"] = runtime.GOOS
 
+	return expandEnvironmentRecursively(responseEnvironment)
+}
+
+func (executor *Executor) PopulateCloneAndWorkingDirEnvironmentVariables(environment map[string]string) map[string]string {
+	result := make(map[string]string)
+	for key, value := range environment {
+		result[key] = value
+	}
+
+	_, hasCloneDir := result["CIRRUS_CLONE_DIR"]
+	_, hasWorkingDir := result["CIRRUS_WORKING_DIR"]
+
 	// Use directory created by the persistent worker if CIRRUS_WORKING_DIR
 	// was not overridden in the task specification by the user
-	_, hasWorkingDir := responseEnvironment["CIRRUS_WORKING_DIR"]
-	if !hasWorkingDir && executor.preCreatedWorkingDir != "" {
-		responseEnvironment["CIRRUS_WORKING_DIR"] = executor.preCreatedWorkingDir
+	if executor.preCreatedWorkingDir != "" {
+		result["CIRRUS_CLONE_DIR"] = executor.preCreatedWorkingDir
+		hasCloneDir = true
+		// Need to override working dir too
+		if hasWorkingDir && !strings.Contains(result["CIRRUS_WORKING_DIR"], "CIRRUS_CLONE_DIR") {
```

Comment:
Not sure if the `$CIRRUS_CLONE_DIR` variable occurrence in the middle of `CIRRUS_WORKING_DIR` should be considered as the `CIRRUS_WORKING_DIR` won't be rooted in `CIRRUS_CLONE_DIR` this way.

See #129 for a more strict check.